### DISCUSSION TITLE
FiPy turbulent flow

### DISCUSSION THREAD

**Discussion by jettrich** [usnistgov/fipy#890] [General]
Dear FiPy Developers/Users,

from an educational point of view I'm currently experimenting on setting up a few basic 1D and 2D examples (mainly in the area of fluid dynamics and heat transfer) with FiPy. Going to more advanced scenarios I also succeeded in doing some simple 3D laminar channel flows, thanks to the examples on the FiPy documentation page and the information here on SO. However, I was asking myself, if anyone ever tried to do a turbulent flow example by implementing a turbulence model?

I'm aware of the fact, that there are a lot of alternatives available - thinking of sophisticated CFD tools like OpenFOAM, Fenics, SU2 and many more. On the other hand, I think that FiPy has some 'charm': i) starting with Python is the lower hurdle for most students, than using C/C++ or Fortran ii) physical Equations are defined quite clear and readable iii) one can start exploring different modeling and simulation strategies without fiddling around with too much programming issues iv) ... and many more, although I am sure that many of you are just waiting to name numerous disadvantages, which certainly exist ;)

However, I would very much appreciate to get into exchange and discussion! Has anyone ever tried to implement a turbulence model, and is anyone aware of a resource that provides a wider range of examples, especially in the areas of fluid flow and heat transfer?

Kind Regards

**Comment by wd15** [comment]
Hi @jettrich: thanks for raising this question. As you mentionied, we (FiPy developers) know of many instances of FiPy being used for educational purposes for the reasons that you've mentioned. Unfortunately, we really don't have any knowledge of people using FiPy to model turbulence. I haven't ever really done any work (or @guyer probably) with turbulence so we can't really help. We'd certainly like to include an example of turbulent flow if you ever get something working.
